x-pack/plugins/security/server/plugin.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { first, map } from 'rxjs/operators';
+import {
+  ClusterClient,
+  CoreSetup,
+  KibanaRequest,
+  Logger,
+  PluginInitializerContext,
+  RecursiveReadonly,
+  deepFreeze,
+} from '../../../../src/core/server';
+import { XPackInfo } from '../../../legacy/plugins/xpack_main/server/lib/xpack_info';
+import { AuthenticatedUser } from '../common/model';
+import { Authenticator, setupAuthentication } from './authentication';
+import { createConfig$ } from './config';
+
+/**
+ * Describes a set of APIs that is available in the legacy platform only and required by this plugin
+ * to function properly.
+ */
+export interface LegacyAPI {
+  xpackInfo: Pick<XPackInfo, 'isAvailable' | 'feature'>;
+  isSystemAPIRequest: (request: KibanaRequest) => boolean;
+}
+
+/**
+ * Describes public Security plugin contract returned at the `setup` stage.
+ */
+export interface PluginSetupContract {
+  authc: {
```

Comment:
__note:__ I wanted to divide/group authc, authz and common exposed API for a long time already :)